src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,55 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task that needs to be done before a specific date.
+ * E.g. submit assignment by Nov-9-2021.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    /**
+     * Class constructor.
+     *
+     * @param description Description of the Deadline.
+     * @param by Date of the Deadline.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDate.parse(by);
+    }
+
+    /**
+     * Returns the date of the deadline formatted as [MMM-dd-yyyy]
+     *
+     * @return The formatted date
+     */
+    public String getDate() {
+        DateTimeFormatter dateFormatObj = DateTimeFormatter.ofPattern("MMM-dd-yyyy");
+        String formattedDate = by.format(dateFormatObj);
+        return formattedDate;
+    }
+
+    /**
+     * Returns formatted string to write to the duke.txt file.
+     *
+     * @return String to write to duke.txt
+     */
+    @Override
+    public String toWrite() {
```

Comment:
I do feel that this method name is a little ambiguous or might be misinterpreted? Might I suggest something like "convertToWriteFormat" or "toWriteFormat"

Reply:
Thanks for the suggestion!